hs_file_types/models/base.py

```diff
@@ -595,3 +646,165 @@ def logical_delete(self, user, delete_res_files=True):
+    dataset_name = models.CharField(max_length=255, null=True, blank=True)
+
+    # this will be used for hsterms:dataType in resourcemetadata.xml
+    # each specific logical type needs to rest this field
+    data_type = "Generic data"
+
+    metadata = models.OneToOneField(BaseFileMetaData, related_name="logical_file")
+
+    @classmethod
+    def create(cls):
+        """this custom method MUST be used to create an instance of this class"""
+        generic_metadata = BaseFileMetaData.objects.create(file_metadata_type='Generic')
+        return cls.objects.create(metadata=generic_metadata)
+
+    @classmethod
+    def get_allowed_uploaded_file_types(cls):
+        # any file can be part of this logical file group - subclass needs to override this
+        return [".*"]
+
+    @classmethod
+    def get_allowed_storage_file_types(cls):
+        # can store any file types in this logical file group - subclass needs to override this
+        return [".*"]
+
+    @classmethod
+    def type_name(cls):
+        return cls.__name__
+
+    @property
+    def has_metadata(self):
+        return hasattr(self, 'metadata')
+
+    @property
+    def size(self):
+        # get total size (in bytes) of all files in this file type
+        return sum([f.size for f in self.files.all()])
+
+    @property
+    def resource(self):
+        res_file = self.files.all().first()
+        if res_file is not None:
+            return res_file.resource
+        else:
+            return None
+
+    @property
+    def supports_resource_file_move(self):
+        """allows a resource file that is part of this logical file type to be moved"""
+        return True
+
+    @property
+    def supports_resource_file_rename(self):
+        """allows a resource file that is part of this logical file type to be renamed"""
+        return True
+
+    @property
+    def supports_zip(self):
+        """allows a folder containing resource file(s) that are part of this logical file type
+        to be zipped"""
+        return True
+
+    @property
+    def supports_delete_folder_on_zip(self):
+        """allows the original folder to be deleted upon zipping of that folder"""
+        return True
+
+    @property
+    def supports_unzip(self):
+        """allows a zip file that is part of this logical file type to get unzipped"""
+        return True
+
+    def add_resource_file(self, res_file):
+        """Makes a ResourceFile (res_file) object part of this logical file object. If res_file
+        is already associated with any other logical file object, this function does not do
+        anything to that logical object. The caller needs to take necessary action for the
+        previously associated logical file object. If res_file is already part of this
+        logical file, it raise ValidationError.
+
+        :param res_file an instance of ResourceFile
+        """
+
+        if res_file in self.files.all():
```

Comment:
Does not work properly because there is nothing preventing one from having two ResourceFile's with the exact same content and different container objects. This will always be false, because there is a new container object and that will differ from the others. Need to check that the file names are different!

Also need to check that you don't add the same physical file to two logical files.....